openmicroscopy/openmicroscopy
# Fixing logging rotation

This PR should resolve issue with logging rotation. It seems that standard logging is thread-safe, but the standard handlers cannot be safely used to log from multiple processes into a single file. Now it was replaced by ```ConcurrentLogHandler``` that allows shared access to the log file. More in https://pypi.python.org/pypi/ConcurrentLogHandler/0.9.1

To test:
 - ssh to trout.o.org
 - ```ls -al /opt/hudson/workspace/OMERO-5.1-breaking-deploy/src/dist/var/log/```
 - start browsing through web and watch how OMEROweb.log is growing. Once is 5MB it should rotate.
Hint: open large dataset and/or tiled image, they produce more logs ;-)

```
-rw-r--r--   1 ola  staff         0  1 Jul 12:43 OMEROweb.lock
-rw-r--r--   1 ola  staff     59957  1 Jul 12:48 OMEROweb.log
-rw-r--r--   1 ola  staff   5243139  1 Jul 12:48 OMEROweb.log.1
-rw-r--r--   1 ola  staff   5242906  1 Jul 12:46 OMEROweb.log.2
```

Note: number of files were extended to 10. Logs will rich max 50MB
cc: @manics @joshmoore @kennethgillen
